operator/src/lib.rs

```diff
@@ -6,71 +6,141 @@ use crate::error::Error;
 use async_trait::async_trait;
 use handlebars::Handlebars;
 use k8s_openapi::api::core::v1::{
-    ConfigMap, ConfigMapVolumeSource, Container, Pod, PodSpec, Volume, VolumeMount,
+    ConfigMap, ConfigMapVolumeSource, Container, Node, Pod, PodSpec, Volume, VolumeMount,
 };
-use kube::api::{ListParams, Meta};
+use kube::api::ListParams;
 use kube::Api;
 use serde_json::json;
-use tracing::{info, trace, warn};
+use tracing::{debug, info, trace, warn};
 
-use stackable_kafka_crd::{KafkaBroker, KafkaCluster, KafkaClusterSpec};
+use stackable_kafka_crd::{KafkaCluster, KafkaClusterSpec};
 use stackable_operator::client::Client;
 use stackable_operator::controller::{Controller, ControllerStrategy, ReconciliationState};
+use stackable_operator::metadata;
 use stackable_operator::reconcile::{
-    ReconcileFunctionAction, ReconcileResult, ReconciliationContext,
+    ContinuationStrategy, ReconcileFunctionAction, ReconcileResult, ReconciliationContext,
 };
-use stackable_operator::{finalizer, metadata, podutils};
-use stackable_zookeeper_crd::{ZooKeeperCluster, ZooKeeperClusterSpec};
+use stackable_zookeeper_crd::ZooKeeperCluster;
 
+use stackable_operator::k8s_utils::LabelOptionalValueMap;
+use stackable_operator::role_utils::RoleGroup;
+use stackable_operator::{k8s_utils, role_utils};
 use std::collections::{BTreeMap, HashMap};
 use std::future::Future;
 use std::pin::Pin;
+use std::string::ToString;
 use std::time::Duration;
+use strum::IntoEnumIterator;
+use strum_macros::Display;
+use strum_macros::EnumIter;
 
-const FINALIZER_NAME: &str = "kafka.stackable.tech/cleanup";
+pub const CLUSTER_NAME_LABEL: &str = "app.kubernetes.io/instance";
+
+pub const NODE_GROUP_LABEL: &str = "kafka.stackable.tech/node-group-name";
+
+pub const NODE_TYPE_LABEL: &str = "kafka.stackable.tech/node-type";
 
 type KafkaReconcileResult = ReconcileResult<error::Error>;
 
+#[derive(EnumIter, Debug, Display, PartialEq, Eq, Hash)]
+pub enum KafkaNodeType {
+    Broker,
+}
+
 struct KafkaState {
     context: ReconciliationContext<KafkaCluster>,
-    kafka_spec: KafkaClusterSpec,
-    zk_spec: Option<ZooKeeperClusterSpec>,
+    kafka_definiton: KafkaCluster,
```

Comment:
Why not just kafka_cluster instead of kafka_definition? Would be more consistent with zk_cluster as well.

Reply:
I remember that I had a reason for this, but looking back I really cannot remember what it was :) 
I'll change it.